packages/flutter_tools/lib/src/build_runner/build_runner.dart

```diff
@@ -114,9 +111,13 @@ class BuildRunner extends CodeGenerator {
         scriptIdFile.createSync(recursive: true);
       }
       scriptIdFile.writeAsBytesSync(appliedBuilderDigest);
-      final PackageGraph packageGraph = PackageGraph.forPath(syntheticPubspec.parent.path);
-      final BuildScriptGenerator buildScriptGenerator = const BuildScriptGeneratorFactory().create(flutterProject, packageGraph);
-      await buildScriptGenerator.generateBuildScript();
+      final ProcessResult generateResult = await processManager.run(<String>[
+        'pub', 'run', 'build_runner', 'generate-build-script'
```

Comment:
So, sometimes we exercise the `builder_runner` API, other times we call `pub`. Does this mean that if a developer adds a `build.yaml` to their project it would apply to some build steps but not others? Could it lead to confusing behavior?

Reply:
if they have a build.yaml I don't think it applies to either project currently. I could update this to copy/link a build.yaml file from the root project, but I'll need to do some experimentation to ensure that works correctly.